pynxtools/dataconverter/readers/ellips/reader.py

```diff
@@ -192,6 +283,87 @@ def data_set_dims(whole_data):
     return unique_angles, counts
 
 
+def parameter_array(whole_data, header, unique_angles, counts):
+    """ User defined variables to produce slices of the whole data set
+
+    """
+    my_data_array = np.empty([
+        len(unique_angles),
+        1,
+        counts[0]
+    ])
+
+    block_idx = [np.int64(0)]
+    index = 0
+    while index < len(whole_data):
+        index += counts[0]
+        block_idx.append(index)
+
+    # derived parameters:
+    # takes last but one column from the right (skips empty columns):
+    data_index = 1
+    temp = whole_data[header["colnames"][-data_index]].to_numpy()[
+        block_idx[-1] - 1].astype("float64")
+
+    while temp * 0 != 0:
```

Comment:
This is always False.

Reply:
not if `temp = nan`. There's probably a better way to write that. Suggestions?

Reply:
Yes, use [`math.isnan`](https://docs.python.org/3/library/math.html#math.isnan)